manifests/config/postgres.pp

```diff
@@ -0,0 +1,29 @@
+# == Define: xylem::config::postgres
+#
+# [*postgres_host*]
+#   Host to create databases on.
+#
+# [*postgres_user*]
+#   User to create databases with.
+#
+# [*postgres_secret*]
+#   Secret key for storing generated credentials.
+#
+# [*postgres_password*]
+#   Optional password for postgres user.
+#
+class xylem::config::postgres(
+  $postgres_host,
+  $postgres_user,
+  $postgres_secret,
+  $postgres_password = undef,
+) {
+  if $postgres_host == undef { fail('postgres_host must be provided') }
+  if $postgres_user == undef { fail('postgres_user must be provided') }
+  if $postgres_secret == undef { fail('postgres_secret must be provided') }
```

Comment:
Just seems to me like we're adding code for something that's a language feature. There are all kinds of places where you could conceivably set a parameter to something it's not supposed to be (e.g. undef) and things would go wrong. This seems like a special case and I haven't seen it elsewhere.